PhysObjectExtractor/src/GenParticleAnalyzer.cc

```diff
@@ -0,0 +1,214 @@
+      std::vector<float> GenPart_pt;
+      std::vector<float> GenPart_eta;
+      std::vector<float> GenPart_mass;
+      std::vector<int> GenPart_pdgId;
+      std::vector<float> GenPart_phi;
+      std::vector<float> GenPart_px;
+      std::vector<float> GenPart_py;
+      std::vector<float> GenPart_pz;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+
+GenParticleAnalyzer::GenParticleAnalyzer(const edm::ParameterSet& iConfig):
+particle(iConfig.getParameter<std::vector<std::string> >("input_particle"))
+
+{
+//now do what ever initialization is needed
+	edm::Service<TFileService> fs;
+	mtree = fs->make<TTree>("Events", "Events");
+    
+	 mtree->Branch("GenPart_pt",&GenPart_pt);
+    mtree->GetBranch("GenPart_pt")->SetTitle("generator particle transverse momentum");
+    mtree->Branch("GenPart_eta",&GenPart_eta);
+    mtree->GetBranch("GenPart_eta")->SetTitle("generator particle pseudorapidity");
+    mtree->Branch("GenPart_mass",&GenPart_mass);
+    mtree->GetBranch("GenPart_mass")->SetTitle("generator particle mass");
+    mtree->Branch("GenPart_pdgId",&GenPart_pdgId);
+    mtree->GetBranch("GenPart_pdgId")->SetTitle("generator particle PDG id");
+    mtree->Branch("GenPart_phi",&GenPart_phi);
+    mtree->GetBranch("GenPart_phi")->SetTitle("generator particle azimuthal angle of momentum vector");
+    mtree->Branch("GenPart_px",&GenPart_px);
+    mtree->GetBranch("GenPart_px")->SetTitle("generator particle x coordinate of momentum vector");
+    mtree->Branch("GenPart_py",&GenPart_py);
+    mtree->GetBranch("GenPart_py")->SetTitle("generator particle y coordinate of momentum vector");
+    mtree->Branch("GenPart_pz",&GenPart_pz);
+    mtree->GetBranch("GenPart_pz")->SetTitle("generator particle z coordinate of momentum vector");
+    mtree->Branch("GenPart_status",&GenPart_status);
+    mtree->GetBranch("GenPart_status")->SetTitle("Particle status. 1=stable");
+
+}
+
+GenParticleAnalyzer::~GenParticleAnalyzer()
+{
+   // do anything here that needs to be done at desctruction time
+   // (e.g. close files, deallocate resources etc.)
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void
+GenParticleAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+   using namespace edm;
+   using namespace std;
+    
+   Handle<reco::GenParticleCollection> gens;
+   iEvent.getByLabel("genParticles", gens);
+   
+  int a1,b1,a2,b2,a3,b3,a4,b4; 
+
+  if(std::find(particle.begin(), particle.end(), "electron") != particle.end())
+  {
+     a1=11;
+     b1=1;
+  }
+  if(std::find(particle.begin(), particle.end(), "muon") != particle.end())
+  {
+     a2=13;
+     b2=1;
+  }
+  if(std::find(particle.begin(), particle.end(), "photon") != particle.end())
+  {
+     a3=22;
+     b3=1;
+  }
+  if(std::find(particle.begin(), particle.end(), "tau") != particle.end())
+  {
+     a4=15;
+     b4=2;
+  }
```

Comment:
So, here, related to the line 78 above, you could read, in a loop, all these a and b directly.  That is the idea.  C++ has some string methods that can be useful